qiskit/visualization/pass_manager_visualization.py

```diff
@@ -193,3 +193,160 @@ def _get_node_color(pss, style):
             return color
 
     return "black"
+
+
+@_optionals.HAS_GRAPHVIZ.require_in_call
+@_optionals.HAS_PYDOT.require_in_call
+def staged_pass_manager_drawer(pass_manager, filename=None, style=None, raw=False):
+    """
+    Draws the staged pass manager.
+
+        This function needs `pydot <https://github.com/erocarrera/pydot>`, which in turn needs
+    Graphviz <https://www.graphviz.org/>` to be installed.
+
+    Args:
+        pass_manager (StagedPassManager): the staged pass manager to be drawn
+        filename (str): file path to save image to
+        style (dict or OrderedDict): keys are the pass classes and the values are
+            the colors to make them. An example can be seen in the DEFAULT_STYLE. An ordered
+            dict can be used to ensure a priority coloring when pass falls into multiple
+            categories. Any values not included in the provided dict will be filled in from
+            the default dict
+        raw (Bool) : True if you want to save the raw Dot output not an image. The
+            default is False.
+    Returns:
+        PIL.Image or None: an in-memory representation of the pass manager. Or None if
+        no image was generated or PIL is not installed.
+    Raises:
+        MissingOptionalLibraryError: when nxpd or pydot not installed.
+        VisualizationError: If raw=True and filename=None.
+
+    Example:
+        .. code-block::
+
+            %matplotlib inline
+            from qiskit.providers.fake_provider import FakeLagosV2
+            from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+
+            pass_manager = generate_preset_pass_manager(3, FakeLagosV2())
+            pass_manager.draw()
+    """
+    import pydot
+
+    # only include stages that have passes
+    stages = list(filter(lambda s: s is not None, pass_manager.expanded_stages))
+
+    if not style:
+        style = DEFAULT_STYLE
+
+    # create the overall graph
+    graph = pydot.Dot()
+
+    # identifiers for nodes need to be unique, so assign an id
+    # can't just use python's id in case the exact same pass was
+    # appended more than once
+    component_id = 0
+
+    # keep a running count of indexes across stages
+    idx = 0
+
+    prev_node = None
+
+    for st in stages:
+        stage = getattr(pass_manager, st)
+
+        if stage is not None:
+            passes = stage.passes()
+            stagegraph = pydot.Cluster(str(st), label="", fontname="helvetica", labeljust="l")
```

Comment:
Could we add the label with the stage name? I think it'd be good to have the stage names in the output visualization.